scripts/check_frontend_test_coverage.py

```diff
@@ -0,0 +1,217 @@
+        'python', '-m', 'scripts.run_frontend_tests'])
+
+
+def create_tmp_folder():
+    """Creates a temporary folder."""
+    if not os.path.exists('./tmp'):
+        os.mkdir('./tmp')
+
+
+def delete_tmp_folder():
+    """Delete the temporary folder."""
+    if os.path.exists('./tmp'):
+        shutil.rmtree('./tmp')
+
+
+def filter_lines(line):
+    """Check if the line has file path (SF) or total lines (LF) or covered
+        lines (LH) of the test.
+
+    Args:
+        line: str. A line from lcov file.
+
+    Returns:
+        Boolean. If the line has the file path or total lines or covered lines
+        of the test.
+    """
+    return (line.find('SF') >= 0 or line.find('LH') >= 0
+            or line.find('LF') >= 0)
+
+
+def get_test_file_name(test_path):
+    """Get the file name from the absolute path.
+
+    Args:
+        test_path: string. The file's absolute path.
+
+    Returns:
+        String. The file path.
+    """
+    if not test_path:
+        sys.stderr.write(
+            'The test path is empty or null.'
+            'It\'s not possible to diff the test coverage correctly.')
+        sys.exit(1)
+
+    return test_path[test_path.rindex('/') + 1:]
+
+
+def get_lcov_file_tests(file_path):
+    """Get all tests from a lcov file, and filters it to return only:
+    - File path
+    - File total lines
+    - File covered lines
+
+    Args:
+      file_path: string. The path of lcov file.
+
+    Returns:
+      String array. An array with all tests filtered, including only important
+      data for the diff.
+    """
+    with python_utils.open_file(file_path, 'r') as f:
+        tests_array = f.read().split('end_of_record')
+        tests_array_filtered = []
+
+        for test in tests_array:
+            lines = [line for line in test.splitlines() if filter_lines(line)]
+            if len(lines) > 0:
+                tests_array_filtered.append(lines)
+
+        return tests_array_filtered
+
+
+def build_tests_fully_coverage_dict():
+    """Build a dict with only fully covered files from develop branch.
+
+    Returns:
+        Dictionary. A dict containing file path, total lines and covered lines
+        of each tested file.
+
+    Raises:
+      Exception: If DEV_LCOV_FILE_PATH doesn't exist.
+    """
+    coverage_dict = {}
+
+    if not os.path.exists(DEV_LCOV_FILE_PATH):
+        raise Exception(
+            'File at path {} doesn\'t exist'.format(DEV_LCOV_FILE_PATH))
+
+    tests = get_lcov_file_tests(DEV_LCOV_FILE_PATH)
+    for lines in tests:
+        total_lines = lines[1].split(':')[1]
+        covered_lines = lines[2].split(':')[1]
+
+        if total_lines == covered_lines:
+            test_name = get_test_file_name(lines[0])
+            coverage_dict[test_name] = [
+                int(total_lines),
+                int(covered_lines)
+            ]
+
+    return coverage_dict        
+
+
+def lcov_files_diff():
+    """Check if any 100% covered file had the coverage dropped.
```

Comment:
Does this handle file renames correctly?

Reply:
Good catch, I don't think so. The file renaming shouldn't be a problem, except if the file is renamed and the coverage of it drops, then I can't track the coverage percentage of this file anymore. Is there a way to analyze the PR commits to check if there was any rename or delete action? What do you think about this approach? It could be a way to fix the gap.

Reply:
I didn't get the whilelist thing, where the whilelist should be in? Can you explain to me more about it?

The omition strategy sounds good, since it won't happen frequently. Can you think of a situation where a file that already is in develop may changes its name?

I think analyzing commits can be very complex, I'll checkout if there is any Python library that has integration with git, it may help.

Reply:
I don't think you'll be able to do rename detection successfully TBH, but feel free to try -- I say this because git doesn't have very good functionality for detecting that, based on past experience.

There's a refactor of the frontend codebase going on now and several files are changing names from camelCase to kebab-case. That might be one example.

You can have the whitelist as a constant in this file -- a list of dirs/regex patterns -- and have the checks run only on files that match patterns in the whitelist.

Reply:
I've found a [Python library](https://github.com/ishepard/pydriller) that analyze git repositories. Do you think worth it to try to implement this lib in the script despite your advice?

I think the file renaming will happen a lot with the Angular migration, you're right, it's an important scenario to think about.

Just a question: if we implement the whilelist, everytime a file gets to 100% coverage, it will be inserted in the constant, right?

Reply:
Please see my comment here, which outlines the whitelist solution in more detail and obviates the need for rename detection or commit analysis: https://github.com/oppia/oppia/pull/8130#discussion_r357045064